### PR TITLE
Replace PassThrough with its child in VP

### PR DESCRIPTION
**Pull request by liqunl**
Replace PassThrough node with its child in its non-NullCheck parents

Signed-off-by: Liqun Liu <liqunl@ca.ibm.com>

### CODE REVIEW

**Comment by liqunl** [comment]
@andrewcraik @jdmpapin Could you review?

File: compiler/optimizer/VPHandlersCommon.cpp [hunk 1]
```diff
@@ -69,8 +70,25 @@ TR::Node *constrainChildren(OMR::ValuePropagation *vp, TR::Node *node)
...
    for (int32_t i = node->getNumChildren()-1; i >= 0; i--)
       {
+      TR::Node* child = node->getChild(i);
       vp->setCurrentParent(node);
-      vp->launchNode(node->getChild(i), node, i);
+      vp->launchNode(child, node, i);
+
+      // Replace PassThrough with its child
+      if (child->getOpCodeValue() == TR::PassThrough
+          && !node->getOpCode().isNullCheck())
```
Comment (jdmpapin) [inline compiler/optimizer/VPHandlersCommon.cpp]: It's possible for `launchNode()` to replace the child, in which case `child` wouldn't be right anymore.

I thought `constrainChildren()` might be able to get away without having to test for null checks, but if not, perhaps this is better done inside `launchNode()`.

Should we use `performTransformation()` here?

File: compiler/optimizer/VPHandlersCommon.cpp [hunk 2]
```diff
@@ -69,8 +70,25 @@ TR::Node *constrainChildren(OMR::ValuePropagation *vp, TR::Node *node)
...
+         {
+         TR::Node* grandChild = child->getFirstChild();
+         node->setAndIncChild(i, grandChild);
+
+         if (child->getReferenceCount() == 1)
+            child->recursivelyDecReferenceCount();
+         else
+            child->decReferenceCount();
```
Comment (jdmpapin) [inline compiler/optimizer/VPHandlersCommon.cpp]: Couldn't this conditional be replaced with an unconditional call to `recursivelyDecReferenceCount()`?

**Comment by liqunl** [comment]
@jdmpapin Changes have been updated.

**Comment by liqunl** [comment]
@andrewcraik Change has passed pbuild.

**Comment by andrewcraik** [comment]
LGTM - @vijaysun-omr over to you

**Comment by vijaysun-omr** [comment]
@genie-omr build all

**Comment by vijaysun-omr** [comment]
Looks ok to me. Will merge if tests pass.

**Pull request merged**